windows/devices/uwb/UwbConnector.cxx

```diff
@@ -23,72 +23,94 @@ namespace uwb
 {
 struct RegisteredCallbackToken
 {
+    /**
+     * @brief Construct a new Registered Callback Token object
+     *
+     * @param deregister the lambda that is passed by copy that handles deregistration.
+     * The lambda assumes that the caller is holding the mutex UwbConnector::m_eventCallbacksGate
+     */
+    RegisteredCallbackToken(std::function<void(RegisteredCallbackToken*)> deregister) :
+        Deregister([this, deregister]() {

```

Comment:
`std::function` is actually fairly heavy-weight, so we should really move this into the lambda:

```suggestion
        Deregister([this, deregister = std::move(deregister)]() {
```